### ISSUE TITLE
Retry ability doesn't work when happening on hook 

### ISSUE THREAD

**Issue by willoliveira-air** [cypress-io/cypress#23255]
### Current behavior

It was possible to verify that the retry ability from the tests is not working for hooks. 
Specially if the fails is happening during the "before" the whole test spec is failing and there is no retry for this instance. 

package.json
`    
"cypress": "10.4.0",
    "cypress-grep": "^2.5.3",
    "@testing-library/cypress": "8.0.3",
    "cy-verify-downloads": "0.1.8",
    "cypress-audit": "^1.1.0",
    "cypress-real-events": "^1.5.1",
    "cypress-terminal-report": "^3.3.2",
`

Message:
`Account does not exist
Because this error occurred during a before all hook we are skipping the remaining tests in the current suite: Freeform annotation actions Although you have test retries enabled, we do not retry tests when before all or after all hooks fail`

### Desired behavior

Retry ability could be able to retry the whole test spec when failing during the hooks. 

### Test code to reproduce

- Add failures during the Before hook 

### Cypress Version

10.4.0

### Other

The retry ability is working fine during the whole test scope, but failing the whole suite before starting the tests it makes difficult to fix the tests.

**Comment by AtofStryker** [comment]
Hi @willoliveira-air. Thank you for opening an issue. Would you be able to provide a reproduction repository to showcase the behavior you expect. When did this behavior change for you?

**Comment by willoliveira-air** [comment]
@AtofStryker thanks for taking a look.

Well, if you add an error (manually) in your before hook you will see the error message for the whole suite. Let's say you have a test spec with 3 tests and you want to visit a page and wait an element to be visible before starting the tests, if this element is not visible, it will fail because it is inside the hook and not in the it context. Cypress wont try to retry with the same message that I reported.

`describe('testing sample', () => {
    before(() => {
      cy.visit(xxxxx);
      cy.get('#xxxx').should('exist').and('be.visible');
    });

    it('should validate 0', () => {
      cy.findByTestId(xxxxx).should('exist').and('be.visible');
    });

    it('should validate 1', () => {
      cy.findByTestId(xxxxx).should('exist').and('be.visible');
    });

    it('should validate 2', () => {
      cy.findByTestId(xxxxx).should('exist').and('be.visible');
    });});});`


Basically I never had this problem before because I was using mock data for my before and when changing for a environment data this error comes up with some changes on fe. I know where the error came from in my suite but I would like to check if this is the expected behaviour from the retry ability or it could be fixed.

`Because this error occurred during a before all hook we are skipping the remaining tests in the current suite: tags
Although you have test retries enabled, we do not retry tests when before all or after all hooks fail`

Thanks

**Comment by willoliveira-air** [comment]
@AtofStryker thanks for taking a look.

Well, if you add an error (manually) in your before hook you will see the error message for the whole suite. Let's say you have a test spec with 3 tests and you want to visit a page and wait an element to be visible before starting the tests, if this element is not visible, it will fail because it is inside the hook and not in the IT context. Cypress won't try to retry with the same message that I reported.


```
describe('testing sample', () => {
    before(() => {
      cy.visit(xxxxx);
      cy.get('#xxxx').should('exist').and('be.visible');
    });
    
    it('should validate 0', () => {
      cy.findByTestId(xxxxx).should('exist').and('be.visible');
    });

    it('should validate 1', () => {
      cy.findByTestId(xxxxx).should('exist').and('be.visible');
    });

    it('should validate 2', () => {
      cy.findByTestId(xxxxx).should('exist').and('be.visible');
    });
  });
});
```

Basically I never had this problem before because I was using mock data on before hook and when changing for an environment data this error comes up with some new changes on fe. I know where the error came from in my suite and one of the changes that I made was moving the failed part from the before to IT context. On that way, I'm still seeing the retry ability when it fails and not failing the whole suite. 


```
describe('testing sample', () => {
    before(() => {
      cy.visit(xxxxx);
    });
    it('should validate 0', () => {
      cy.get('#xxxx').should('exist').and('be.visible');
      cy.findByTestId(xxxxx).should('exist').and('be.visible');
    });
    it('should validate 1', () => {
      cy.get('#xxxx').should('exist').and('be.visible');
      cy.findByTestId(xxxxx).should('exist').and('be.visible');
    });
    it('should validate 2', () => {
      cy.get('#xxxx').should('exist').and('be.visible');
      cy.findByTestId(xxxxx).should('exist').and('be.visible');
    });
   });
});
```

I would like to check if this is the expected behaviour from the retry ability or it could be fixed.

```
Because this error occurred during a before all hook we are skipping the remaining tests in the current suite: tags
Although you have test retries enabled, we do not retry tests when before all or after all hooks fail
```

Thanks

**Comment by AtofStryker** [comment]
@willoliveira-air I think I know what you mean. In other words you want the `before` hook to retry when failed. Currently this is expected behavior. I put together a [reproduction repo](https://github.com/AtofStryker/cypress-issue-23255) that goes into what you described. I also provided a work around there, but it does go against what we would [recommend](https://docs.cypress.io/guides/references/best-practices#Having-tests-rely-on-the-state-of-previous-tests). Would you like me to label this ask a feature request?

**Comment by willoliveira-air** [comment]
Hey @AtofStryker got it. Actually it doesn't go against what you recommend. For that particular case on the before hook I'm creating the env data with ephemeral environment and then the tests specs are running with shared code on its before each. 

It is pretty much similar from what you call as: "Run shared code before each test" but in this case, it could be shared "data" and this is pretty much one extension from what we have in before each for the before context. However, I got your point and no problem.  

I would love to see it as feature request. Could you please tag?
Thank you so much for taking a looking and debugging it properly.

**Comment by vijaybtech13** [comment]
Hi Team - Is retry ability supported for before hook failures?
